src/Core/Framework/DataAbstractionLayer/Dbal/Common/IterableQuery.php

```diff
@@ -23,7 +21,7 @@ public function fetchCount(): int;
     public function getQuery(): QueryBuilder;
 
     /**
-     * @return Offset
```

Comment:
can we split this from the other changes? Those are two different things, and i wanna understand the root cause here, as in theory the doc annotations should have no effect for serialization where we type hint array natively

Also this change is again easy to revert and thus accidentally breaking serialization again